src/store/modules/time.ts

```diff
@@ -0,0 +1,87 @@
+import {TimeObject, TimeState} from "@/store/interfaces";
```

Comment:
just FYI: im pretty sure you would find a third party component with this functionality, cool solution either way